cmd/github-ldap-user-group-creator/main.go

```diff
@@ -142,12 +148,60 @@ func main() {
 		logrus.WithError(err).Fatal("Failed to load the mapping")
 	}
 
-	if err := ensureGroups(ctx, clients, mapping, opts.dryRun); err != nil {
+	data, err := ioutil.ReadFile(opts.groupsFile)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to read the group file")
+	}
+	roverGroups := map[string][]string{}
+	if err := yaml.Unmarshal(data, &roverGroups); err != nil {
+		logrus.WithError(err).Fatal("Failed to unmarshal groups")
+	}
+
+	if ciAdmins, ok := roverGroups[api.CIAdminsGroupName]; !ok {
+		logrus.WithField("groupName", api.CIAdminsGroupName).Fatal("Failed to find ci-admins group")
+	} else if l := len(ciAdmins); l < 3 {
+		logrus.WithField("groupName", api.CIAdminsGroupName).WithField("len", l).Fatal("Require at least 3 members of ci-admins group")
+	}
+
+	groups := map[string]*userv1.Group{}
+	for k, v := range makeGitHubMappingGroups(mapping) {
+		groups[k] = v
+	}
+	for k, v := range makeRoverGroups(roverGroups) {
+		if _, ok := groups[k]; ok {
+			logrus.WithField("group", k).Fatal("Group has been defined already")
+		}
+		groups[k] = v
+	}
+
+	if err := ensureGroups(ctx, clients, groups, opts.dryRun); err != nil {
 		logrus.WithError(err).Fatal("could not ensure groups")
 	}
 }
 
-func ensureGroups(ctx context.Context, clients map[string]ctrlruntimeclient.Client, mapping map[string]string, dryRun bool) error {
+func makeGitHubMappingGroups(mapping map[string]string) map[string]*userv1.Group {
+	groups := map[string]*userv1.Group{}
+	for githubLogin, kerberosId := range mapping {
+		groupName := fmt.Sprintf("%s%s", githubLogin, api.GroupSuffix)
```

Comment:
this contradicts with your README. I would expect it to be 
```suggestion
		groupName := fmt.Sprintf("%s-%s", githubLogin, api.GroupSuffix)
```

Reply:
https://github.com/openshift/ci-tools/blob/02d9d4fac19ef0184c322ff9a261521f5a643782/pkg/api/constant.go#L57

Actually, the hyphen is a part of `api.GroupSuffix`.
Let me remove the confusion.

Reply:
Updated.